src/ripple/protocol/PublicKey.h

```diff
@@ -131,6 +131,12 @@ class PublicKey
     }
 };
 
+static_assert(std::is_default_constructible_v<PublicKey>);
+static_assert(std::is_copy_constructible_v<PublicKey>);
+static_assert(std::is_move_constructible_v<PublicKey>);
```

Comment:
Is it a good practise to check for move constructibility? PublicKey class only has the compiler provided default move constructor.

Reply:
Speaking only for myself, I think it's often good practice to check at compile time whether a given class has the expected interfaces, regardless of whether they are coded explicitly or provided implicitly by the compiler.  If a user defines copy, move, and/or assignment this can have unexpected consequences unless you (and the readers of your code) know C++ backwards and forwards.  `static_asserts` like these provide an easy compile-time way to see if you have the interface you expect.  They also provide a flag in case someone unintentionally changes that interface in the future.

Reply:
ok, thanks for the clarification!!